gcloud/storage/acl.py

```diff
@@ -345,78 +346,84 @@ def get_entities(self):
         self._ensure_loaded()
         return list(self.entities.values())
 
-    def reload(self):
-        """Reload the ACL data from Cloud Storage.
+    @property
+    def connection(self):
+        """Compute the connection for API requests for this ACL.
 
         This is a virtual method, expected to be implemented by subclasses.
 
-        :raises: :class:`NotImplementedError`
+        :raises: :class:`NotImplementedError` if ``_connection`` attribute
+                 is not set on the instance.
+
         """
+        # Allow override for testing
+        connection = getattr(self, '_connection', None)
```

Comment:
This PR moves the implementations of `reload`, `save`, and `clear` up to the base `ACL` class.  Without having some means of setting a connection, it is not possible to test that the default case (no connection passed) works correctly.

I just checked:  #843 actually fails to test the "no connection passed" case for most methods.